src/storage/src/table/cell_based_table.rs

```diff
@@ -456,3 +464,69 @@ impl<S: StateStore> TableIter for CellBasedTableRowIter<S> {
         }
     }
 }
+
+/// "`CellBasedTableStreamingIter`" used for streaming executor, and will not wait epoch
+pub struct CellBasedTableStreamingIter<S: StateStore> {
+    /// An iterator that returns raw bytes from storage.
+    iter: StripPrefixIterator<S::Iter>,
+    /// Cell-based row deserializer
+    cell_based_row_deserializer: CellBasedRowDeserializer,
+    pub cell_based_item: Option<(Vec<u8>, Row)>,
+}
+
+impl<S: StateStore> CellBasedTableStreamingIter<S> {
+    pub async fn new(
+        keyspace: &Keyspace<S>,
+        table_descs: &[ColumnDesc],
+        epoch: u64,
+    ) -> StorageResult<Self> {
+        let cell_based_row_deserializer = CellBasedRowDeserializer::new(table_descs.to_vec());
+        let iter = keyspace.iter(epoch).await?;
+        let iter = Self {
+            iter,
+            cell_based_row_deserializer,
+            cell_based_item: None,
+        };
+        Ok(iter)
+    }
+
+    pub async fn init_cell_based_item(&mut self) -> StorageResult<()> {
+        if self.cell_based_item.is_none() {
+            self.cell_based_item = self.next().await?;
+        }
+        Ok(())
+    }
+
+    pub async fn update_current_item(&mut self) -> StorageResult<()> {
```

Comment:
Rust-style iterators never have something like `update_current_item`.

next -> return the next item with ownership
peek -> return the current item of reference

Reply:
`next` is not even `pub` -- how can `HashAgg` use such iterator?